### ISSUE TITLE
Separate "User Management" and "Facility Management" Roles

### ISSUE THREAD

**Issue by donhead** [samilliken/openDCIM#766]
We're looking at implementing openDCIM in a fairly large environment, where those responsible for application support and user management may not necessarily be the same folks responsible for managing and maintaining the facilities and other "content" within openDCIM.

Currently, the role of "Manage Site and Users" is required in order to make facility adjustments, and is also utilized for user management.  We'd prefer to see all of the User and Department Administration functions moved under a separate role.

While I understand this may not sound like a huge deal, in an environment where audits of users and permissions are done on a regular basis, it's quite a big deal for us. =)

Thank you!

**Comment by wilpig** [comment]
It is a big deal and a giant pain in the ass to move permissions around like that even though it's on our roadmap.

**Comment by donhead** [comment]
Yeah, I know.  I figured I'd throw this out there as a formal enhancement request.  If you can make it happen, we'd really appreciate it (along with anyone else that deals with various regulatory compliance bodies, such as PCI, FedRAMP, HIPAA, etc).  Until then, it has to go into the bucket of systems where we have to maintain much tighter controls.

**Comment by samilliken** [comment]
I'm thinking I already did this when I added the LDAP integration of 4.2 - Contact Admins can manage departments, users, and the relationships.  I'll check, though.

**Comment by donhead** [comment]
The issue I ran into is that my datacenter manager wanted to add/edit datacenter/facility items, but in order to do so, I had to make him the "Manage Site and Users" role, which means he is suddenly able to create users.  He is not in a position that should be making user changes.  I guess I should have clarified that a bit, sorry about that.

Ideally, we'd have:

- People creating "content" in the system (datacenters, cabinets, racks, etc) to NOT be "full admins"
- A role just for user administration

Does that make sense?

**Comment by donhead** [comment]
Just digging into this quickly, since I have a moment..

Within "User Administration", I have the following options for a given user:

Admin Own Devices
Read/Report Access (Global)
Modify/Enter Devices (Global)
Delete Devices (Global)
Enter/Modify Contacts and Departments
Enter Rack Requests
Complete Rack Requests
Manage Site and Users
Disabled


The role that's bugging me is "Manage Site and Users", since that's what is required in order to edit Datacenter/Infrastructure items.

If that could be split out, here's how I'd love to see it.

Manage Infrastructure
Manage Users
Manage openDCIM Configuration


I could give our "datacenter manager" the "Manage Infrastructure" role, and he could do whatever he wanted with content, without having access to users, or the configuration of openDCIM itself.

I could give our "access administration" team the "Manage Users" role, and they could provision users and do audits and whatnot.

I could give our "application support team" the "Manage openDCIM Configuration" role, and they could go in and tweak the internals.

In smaller shops, it could be a single person fulfills one or more of those roles, but within our larger organization (and others I've worked in previously), there's usually a pretty clear separation of duties along the lines I've defined above.

My 2¢, food for thought, etc.  Curious what your thoughts are.
